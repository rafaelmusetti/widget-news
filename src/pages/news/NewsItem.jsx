import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import moment from 'moment';
import { Row, Col } from 'antd';

const NewsItemContent = styled.div`
  border-bottom: 1px solid;
  padding: 12px 0;
`;

const Title = styled.div`
  font-size: 24px;
  font-weight: 500;
  margin-bottom: 8px;
`;

const PublicationDate = styled.div`
  font-size: 20px;
  color: #A4A4A4;
`;

const Source = styled.span`
  font-size: 20px;
  padding: 0px 12px 4px;
  background-color: #D8D8E4;
  color: #FFFFFF;
`;

const LinkTitle = styled.a`
  color: #4E4E4E;
`;

const NewsItem = ({
  title,
  publishedAt,
  source,
  url,
}) => (
  <NewsItemContent className="news-item">
    <Row>
      <Col span={24}>
        <Title>
          <LinkTitle href={url}>{title}</LinkTitle>
        </Title>
      </Col>
    </Row>
    <Row>
      <Col xs={{ span: 8 }} lg={{ span: 3 }}>
        <PublicationDate className="news-item__date">{moment(publishedAt).format('DD/MM/YYYY')}</PublicationDate>
      </Col>
      <Col xs={{ span: 16 }} lg={{ span: 21 }}>
        <Source className="news-item__source">{source}</Source>
      </Col>
    </Row>
  </NewsItemContent>
);

NewsItem.propTypes = {
  title: PropTypes.string,
  publishedAt: PropTypes.string,
  source: PropTypes.string,
  url: PropTypes.string,
};

NewsItem.defaultProps = {
  title: '',
  publishedAt: '',
  source: '',
  url: '',
};

export default NewsItem;
