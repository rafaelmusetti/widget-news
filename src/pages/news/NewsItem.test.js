import React from 'react';
import { shallow } from 'enzyme';
import NewsItem from './NewsItem';

describe('<NewsItem />', () => {
  describe('When rendering', () => {
    const tree = shallow(
      <NewsItem
        title="title"
        publishedAt="2019-02-15T14:53:40Z"
        source="source"
        url="url"
      />,
    );

    it('should renders properly', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
