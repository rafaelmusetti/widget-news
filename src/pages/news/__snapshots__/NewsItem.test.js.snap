// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsItem /> When rendering should renders properly 1`] = `
<styled.div
  className="news-item"
>
  <Row
    gutter={0}
  >
    <Col
      span={24}
    >
      <styled.div>
        <styled.a
          href="url"
        >
          title
        </styled.a>
      </styled.div>
    </Col>
  </Row>
  <Row
    gutter={0}
  >
    <Col
      lg={
        Object {
          "span": 3,
        }
      }
      xs={
        Object {
          "span": 8,
        }
      }
    >
      <styled.div
        className="news-item__date"
      >
        15/02/2019
      </styled.div>
    </Col>
    <Col
      lg={
        Object {
          "span": 21,
        }
      }
      xs={
        Object {
          "span": 16,
        }
      }
    >
      <styled.span
        className="news-item__source"
      >
        source
      </styled.span>
    </Col>
  </Row>
</styled.div>
`;
